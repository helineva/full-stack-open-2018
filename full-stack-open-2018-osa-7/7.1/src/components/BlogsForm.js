import React from 'react'
import Blog from './Blog'
import CreateBlogForm from './CreateBlogForm'

class BlogsForm extends React.Component {
  render() {
    return (
      <div className='blogsForm'>
        <h2>blogs</h2>
          {this.props.blogs.map(blog =>
            <Blog
              key={blog._id}
              className={blog}
              blog={blog}
              updateBlog={this.props.updateBlog}
              deleteBlog={this.props.deleteBlog}
              showDeleteButton={blog.user === undefined
                || blog.user._id === this.props.user.id }
            />
          )}
          <CreateBlogForm addBlog={this.props.addBlog}/>
      </div>
    )
  }
}

export default BlogsForm
