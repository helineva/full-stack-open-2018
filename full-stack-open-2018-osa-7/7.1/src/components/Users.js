import React from 'react'
import userService from '../services/users.js'

class Users extends React.Component {
  constructor(props) {
    super(props);
    this.state = {
      users: []
    }
  }

  componentDidMount() {
    userService.getAll().then(users => {
      this.setState({ users });
    });
  }

  render() {
    return (
      <div>
        <h2>users</h2>
          <table>
            <thead>
              <tr>
                <th>user</th>
                <th>blogs added</th>
                <th>adult</th>
              </tr>
            </thead>
            <tbody>
              {this.state.users.map(user => {
                return (
                  <tr key={user._id}>
                    <td>{user.name}</td>
                    <td>{user.blogs.length}</td>
                    <td>{user.adult ? 'yes' : 'no'}</td>
                  </tr>
                )})
              }
            </tbody>
          </table>
      </div>
    )
  }
}

export default Users
