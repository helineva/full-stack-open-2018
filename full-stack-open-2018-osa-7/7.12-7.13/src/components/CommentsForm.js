import React from 'react'
import { notify } from '../reducers/notificationReducer'
import { addComment } from '../reducers/blogReducer'
import { connect } from 'react-redux'
import { Button, Form, List } from 'semantic-ui-react'

class CommentsForm extends React.Component {
  constructor(props) {
    super(props)
    this.state = {
      content: ''
    }
  }

  add = (event) => {
    event.preventDefault();
    if (!this.state.content) {
      return
    }
    try {
      this.props.addComment(this.state.content)
      const msg = `comment '${this.state.content}' added`
      this.props.notify(msg, 'note', 5)
      this.setState({ content: '' })
    } catch (exception) {
      console.log(exception)
    }
  }

  handleFieldChange = (event) => {
    this.setState({ content: event.target.value })
  }

  render() {
    if (this.props.blog) {
      return (
        <div style={{ marginTop: 20 }}>
          <h3>Comments</h3>
          <List bulleted>
            {this.props.blog.comments.map((comment, index) =>
              <List.Item key={index}>{comment}</List.Item>
            )}
          </List>
          <Form onSubmit={this.add}>
            <input
              type="text"
              value={this.state.content}
              onChange={this.handleFieldChange}
            />
            <Button type="submit">add comment</Button>
          </Form>
        </div>
      )
    }
    return null
  }
}

const mapStateToProps = (state, props) => {
  if (!state.blogs) {
    return { blog: null }
  }
  const blog = state.blogs.filter(blog =>
    blog._id === props.blogId)[0]
  return { blog }
}

const mapDispatchToProps = (dispatch, props) => {
  return {
    notify: (msg, cl, time) => dispatch(notify(msg, cl, time)),
    addComment: (content) => dispatch(addComment(props.blogId, content))
  }
}

export default connect(
  mapStateToProps,
  mapDispatchToProps)(CommentsForm);
