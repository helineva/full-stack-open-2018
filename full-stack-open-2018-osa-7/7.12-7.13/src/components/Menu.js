import React from 'react'
import { NavLink } from 'react-router-dom'
import { Button } from 'semantic-ui-react'

const Menu = ({ user, logout }) => {
  const style = {
    backgroundColor: 'lightGrey',
    borderRadius: 5,
    padding: 10,
    marginBottom: 20
  }
  const activeStyle = {
    fontWeight: 'bold'
  }
  return (
    <div style={style}>
      <NavLink to="/blogs" activeStyle={activeStyle}>blogs</NavLink>&emsp;
      <NavLink to='/users' activeStyle={activeStyle}>users</NavLink>&emsp;
      <span>{user.name} logged in&emsp;
        <Button onClick={logout}>logout</Button>
      </span>
    </div>
  )
}

export default Menu
