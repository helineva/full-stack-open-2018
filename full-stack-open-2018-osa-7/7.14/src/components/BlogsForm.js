import React from 'react';
import CreateBlogForm from './CreateBlogForm';
import { connect } from 'react-redux';
import { Link } from 'react-router-dom';
import { Table } from 'semantic-ui-react';

class BlogsForm extends React.Component {
  render() {
    if (this.props.blogs) {
      return (
        <div>
          <h2>blogs</h2>
          <Table striped>
            <Table.Header>
              <Table.Row>
                <Table.HeaderCell>Title</Table.HeaderCell>
                <Table.HeaderCell>Author</Table.HeaderCell>
              </Table.Row>
            </Table.Header>
            <Table.Body>
              {this.props.blogs.map(blog =>
                <Table.Row key={blog._id}>
                  <Table.Cell>
                    <Link to={`/blogs/${blog._id}`}>{blog.title}</Link>
                  </Table.Cell>
                  <Table.Cell>
                    {blog.author}
                  </Table.Cell>
                </Table.Row>
              )}
            </Table.Body>
          </Table>
          <CreateBlogForm />
        </div>
      );
    }
    return null;
  }
}

const mapStateToProps = (state) => {
  return { blogs: state.blogs };
};

export default connect(mapStateToProps)(BlogsForm);
