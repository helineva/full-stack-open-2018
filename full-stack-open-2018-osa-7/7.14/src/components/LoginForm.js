import React from 'react';
import { notify } from '../reducers/notificationReducer';
import { login } from '../reducers/loginReducer';
import { connect } from 'react-redux';
import { Form, Button } from 'semantic-ui-react';

class LoginForm extends React.Component {
  constructor(props) {
    super(props);
    this.state = {
      username: '',
      password: '',
      loginFormVisible: false
    };
  }

  login = (event) => {
    event.preventDefault();
    this.props.login(this.state.username, this.state.password);
  }

  handleFieldChange = (event) => {
    this.setState({ [event.target.name]: event.target.value });
  }

  render() {
    if (this.state.loginFormVisible) {
      return (
        <div>
          <h2>Log in to application</h2>
          <Form onSubmit={this.login}>
            <Form.Field>
              <label>username</label>
              <input
                type="text"
                name="username"
                value={this.state.username}
                onChange={this.handleFieldChange}
              />
            </Form.Field>
            <Form.Field>
              <label>password</label>
              <input
                type="password"
                name="password"
                value={this.state.password}
                onChange={this.handleFieldChange}
              />
            </Form.Field>
            <Button type="submit">login</Button>
            <Button
              type="submit"
              onClick={e => {
                e.preventDefault();
                this.setState({ loginFormVisible: false });}}>
              cancel
            </Button>
          </Form>
        </div>
      );
    }
    return (
      <div>
        <Button onClick={e => this.setState({ loginFormVisible: true })}>
          login
        </Button>
      </div>
    );
  }
}

export default connect(
  null,
  { notify, login })(LoginForm);
