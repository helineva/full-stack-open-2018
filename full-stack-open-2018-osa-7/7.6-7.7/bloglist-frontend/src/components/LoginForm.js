import React from 'react';
import loginService from '../services/login'

class LoginForm extends React.Component {
  constructor(props) {
    super(props);
    this.state = {
      username: '',
      password: '',
      loginFormVisible: false
    }
  }

  login = async (event) => {
    event.preventDefault()
    try {
      const user = await loginService.login({
        username: this.state.username,
        password: this.state.password
      })
      this.props.setUser(user)
    } catch (exception) {
      this.props.setNotification('invalid username or password', 'error')
    }
  }

  handleFieldChange = (event) => {
    this.setState({ [event.target.name]: event.target.value })
  }

  render() {
    if (this.state.loginFormVisible) {
      return (
        <div className='loginForm'>
          <h2>Log in to application</h2>
          <form onSubmit={this.login}>
            <div>
              username:
              <input
                type="text"
                name="username"
                value={this.state.username}
                onChange={this.handleFieldChange}
              />
            </div>
            <div>
              password:
              <input
                type="password"
                name="password"
                value={this.state.password}
                onChange={this.handleFieldChange}
              />
            </div>
            <button type="submit">login</button>
          </form>
          <button onClick={e => this.setState({ loginFormVisible: false })}>
            cancel
          </button>
        </div>
      )
    }
    return (
      <div className='loginForm'>
        <button onClick={e => this.setState({ loginFormVisible: true })}>
          login
        </button>
      </div>
    )
  }
}

export default LoginForm;
