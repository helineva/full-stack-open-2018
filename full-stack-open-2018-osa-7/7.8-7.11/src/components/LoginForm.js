import React from 'react';
import { notify } from '../reducers/notificationReducer'
import { login } from '../reducers/loginReducer'
import { connect } from 'react-redux'

class LoginForm extends React.Component {
  constructor(props) {
    super(props);
    this.state = {
      username: '',
      password: '',
      loginFormVisible: false
    }
  }

  login = (event) => {
    event.preventDefault()
    this.props.login(this.state.username, this.state.password)
  }

  handleFieldChange = (event) => {
    this.setState({ [event.target.name]: event.target.value })
  }

  render() {
    if (this.state.loginFormVisible) {
      return (
        <div className='loginForm'>
          <h2>Log in to application</h2>
          <form onSubmit={this.login}>
            <div>
              username:
              <input
                type="text"
                name="username"
                value={this.state.username}
                onChange={this.handleFieldChange}
              />
            </div>
            <div>
              password:
              <input
                type="password"
                name="password"
                value={this.state.password}
                onChange={this.handleFieldChange}
              />
            </div>
            <button type="submit">login</button>
          </form>
          <button onClick={e => this.setState({ loginFormVisible: false })}>
            cancel
          </button>
        </div>
      )
    }
    return (
      <div className='loginForm'>
        <button onClick={e => this.setState({ loginFormVisible: true })}>
          login
        </button>
      </div>
    )
  }
}

export default connect(
  null,
  { notify, login })(LoginForm);
