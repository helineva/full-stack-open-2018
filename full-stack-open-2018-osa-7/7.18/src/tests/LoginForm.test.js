import React from 'react';
import { shallow } from 'enzyme';
import { LoginForm } from '../components/LoginForm';
import { Form, Button } from 'semantic-ui-react';
import toJson from 'enzyme-to-json';
jest.mock('../services/login');
jest.mock('../reducers/loginReducer');

describe('<LoginForm />', () => {
  it('login form is not shown at first', () => {
    const loginFormComponent = shallow(<LoginForm />);
    const form = loginFormComponent.find(Form);
    expect(form.length).toBe(0);
  });

  it('login form is shown after clicking', () => {
    const loginFormComponent = shallow(<LoginForm />);
    const loginButton = loginFormComponent.find(Button);
    expect(loginButton.length).toBe(1);
    loginButton.simulate('click');
    const form = loginFormComponent.find(Form);
    expect(form.length).toBe(1);
  });

  it('login form is not shown after clicking twice', () => {
    const loginFormComponent = shallow(<LoginForm />);
    const loginButton = loginFormComponent.find(Button);
    expect(loginButton.length).toBe(1);
    loginButton.simulate('click');
    const buttons = loginFormComponent.find(Button);
    expect(buttons.length).toBe(2);
    buttons.at(1).simulate('click', {
      preventDefault: () => {}
    });
    const form = loginFormComponent.find(Form)
    expect(form.length).toBe(0);
  });

  it('passes snapshot test before clicking', () => {
    const loginFormComponent = shallow(<LoginForm />);
    expect(toJson(loginFormComponent)).toMatchSnapshot();
  })

  it('passes snapshot test after clicking', () => {
    const loginFormComponent = shallow(<LoginForm />);
    const loginButton = loginFormComponent.find(Button);
    loginButton.simulate('click');
    expect(toJson(loginFormComponent)).toMatchSnapshot();
  })
});
