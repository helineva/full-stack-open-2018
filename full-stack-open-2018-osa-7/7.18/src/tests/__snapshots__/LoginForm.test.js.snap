// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<LoginForm /> passes snapshot test after clicking 1`] = `
<div>
  <h2>
    Log in to application
  </h2>
  <Form
    as="form"
    onSubmit={[Function]}
  >
    <FormField>
      <label>
        username
      </label>
      <input
        name="username"
        onChange={[Function]}
        type="text"
        value=""
      />
    </FormField>
    <FormField>
      <label>
        password
      </label>
      <input
        name="password"
        onChange={[Function]}
        type="password"
        value=""
      />
    </FormField>
    <Button
      as="button"
      role="button"
      type="submit"
    >
      login
    </Button>
    <Button
      as="button"
      onClick={[Function]}
      role="button"
      type="submit"
    >
      cancel
    </Button>
  </Form>
</div>
`;

exports[`<LoginForm /> passes snapshot test before clicking 1`] = `
<div>
  <Button
    as="button"
    onClick={[Function]}
    role="button"
  >
    login
  </Button>
</div>
`;
